hazelcast/src/test/java/com/hazelcast/cache/impl/AbstracRecordStoreTest.java

```diff
@@ -0,0 +1,106 @@
+package com.hazelcast.cache.impl;
+
+import com.hazelcast.cache.CacheFromDifferentNodesTest;
+import com.hazelcast.cache.CacheTestSupport;
+import com.hazelcast.cache.HazelcastCacheManager;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.TestUtil;
+import com.hazelcast.test.AssertTask;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import org.junit.Test;
+
+import javax.cache.Cache;
+import javax.cache.configuration.FactoryBuilder;
+import javax.cache.configuration.MutableCacheEntryListenerConfiguration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class AbstracRecordStoreTest extends CacheTestSupport {
+
+    TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
+
+    @Test
+    public void testRemovingSameEntryTwiceShouldTriggerEntryListenerOnlyOnce() {
```

Comment:
there is already same test in `CacheBasicAbstractTest` which works for both client and server side caches